### PR TITLE
Docstrings

### LINKED ISSUE #99: Standardize docstrings
Make sure all docstrings adhere to the [NumPy style](https://numpydoc.readthedocs.io/en/latest/format.html).

### PR DESCRIPTION
**Pull request by Spayralbe**
- Add NumPy docstrings to all public modules, methods and classes (Closes #99)
- Make `schema_translate_map` immutable
- Restructure `VocabManager` to call vocab loading from composition classes directly
- Add docstring linter to nox

### CODE REVIEW

File: src/delphyne/database/database.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
+"""Database operations module."""
```
Comment (alepev) [inline src/delphyne/database/database.py:1]: Do we need to add strings like this at the top of every file (including `__init__`s) from now on?

File: src/delphyne/database/database.py [hunk 2]
```diff
@@ -20,10 +23,32 @@
...
+
+    Attributes
+    ----------
+    schemas
+    reflected_metadata
+    engine : sqlalchemy.engine.base.Engine
+        Database engine.
+    constraint_manager : ConstraintManager
+        Access point to alter constraints/indexes of the database.
+    """
```
Comment (alepev) [inline src/delphyne/database/database.py:46]: I think I've seen `Parameters` described for all classes; not sure if it's always the case for `Attributes`, and if/when it's really needed. Do you follow specific guidelines for this? Just curious to know if I should document them in the future

**Review by alepev — APPROVED** [review]

File: src/delphyne/database/database.py (hunk 1 above)
Comment (Spayralbe) [reply src/delphyne/database/database.py:1]: Yes, or the linter police will know where to find you.

File: src/delphyne/database/database.py (hunk 2 above)
Comment (Spayralbe) [reply src/delphyne/database/database.py:46]: I understand it as any public attribute that is not part of the `init`(those are already described under `Parameters`)
[https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring](https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring )

**Pull request merged**